client/my-sites/site-settings/navigation-back-button/index.jsx

```diff
@@ -0,0 +1,33 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { localize } from 'i18n-calypso';
+import Gridicon from 'gridicons';
+import page from 'page';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+
+const NavigationBackButton = ( { redirectRoute, translate } ) => {
+	const handleClick = () => {
```

Comment:
How about using a `Component` or a `PureComponent` instead of a stateless functional component here? The problem with this approach is that it will generate a new `handleClick` handler on every render.

Reply:
Let's avoid `page` routing altogether when there's an `href` prop we can use: https://github.com/Automattic/wp-calypso/blob/master/client/components/button/README.md#props